plugins/docker_binary_builder/README.md

```diff
@@ -0,0 +1,63 @@
+# Docker Builder for Binaries
+
+This plugin is useful when your projects are based on compiled languages and you don't want to base
+your deployment docker images on images that have all the tools/libs/compilers installed.
+
+Advantages to separating your docker images include:
+- Smaller deployment docker images.
+- Better security in your docker images to not be able to recompile code.
+
+## Getting Started
+
+This plugin makes a few assumptions:
+- You have 2 dockerfiles:
+ - 'Dockerfile' used to generate your main deployment docker image
+ - 'Dockerfile.build' used to create a temporary image to build any binaries you want to package into the former.
+- You have a build script 'build.sh' that can be executed to compile the needed binaries
+  - This is packaged inside the Dockerfile.build image and executed
```

Comment:
I don't know about the API itself, but if you call `docker inspect` on a container, you can find out whether it's still running, and what its exit code is.  I *believe* that's the exit code of the CMD that is run.

```
$ docker inspect MYSQL_DATA_NEW | jq '.[0].State'
{
  "Running": false,
  "Paused": false,
  "Restarting": false,
  "OOMKilled": false,
  "Dead": false,
  "Pid": 0,
  "ExitCode": 0,
  "Error": "",
  "StartedAt": "2015-08-28T22:58:21.370347184Z",
  "FinishedAt": "2015-08-28T22:58:21.406439187Z"
}
```

I assume the same information can be retrieved via the API.